src/starknet_runner/runner.rs

```diff
@@ -0,0 +1,388 @@
+            fact_state::{
+                contract_state::ContractState, in_memory_state_reader::InMemoryStateReader,
+                state::ExecutionResourcesManager,
+            },
+            state::{cached_state::CachedState, state_api_objects::BlockInfo},
+        },
+        core::syscalls::syscall_handler::{SyscallHandler, SyscallHintProcessor},
+        definitions::{
+            constants::TRANSACTION_VERSION,
+            general_config::{self, StarknetGeneralConfig},
+        },
+        services::api::contract_class::{ContractClass, ContractEntryPoint, EntryPointType},
+        starknet_storage::dict_storage::DictStorage,
+        utils::{test_utils::vm, Address},
+    };
+
+    use super::StarknetRunner;
+
+    #[test]
+    fn get_execution_resources_test_fail() {
+        let program = cairo_rs::types::program::Program::default();
+        let cairo_runner = CairoRunner::new(&program, "all", false).unwrap();
+        let mut vm = VirtualMachine::new(true);
+        let hint_processor = SyscallHintProcessor::new_empty();
+
+        let runner = StarknetRunner::new(cairo_runner, vm, hint_processor);
+
+        assert!(runner.get_execution_resources().is_err());
+    }
+
+    #[test]
+    fn prepare_os_context_test() {
+        let program = cairo_rs::types::program::Program::default();
+        let cairo_runner = CairoRunner::new(&program, "all", false).unwrap();
+        let mut vm = VirtualMachine::new(true);
+        let hint_processor = SyscallHintProcessor::new_empty();
+
+        let mut runner = StarknetRunner::new(cairo_runner, vm, hint_processor);
+
+        let os_context = runner.prepare_os_context();
+
+        // is expected to return a pointer to the first segment as there is nothing more in the vm
+        let expected = Vec::from([MaybeRelocatable::from((0, 0))]);
+
+        assert_eq!(os_context, expected);
+    }
+
+    #[test]
+    fn integration_test() {
+        // ---------------------------------------------------------
+        //  Create program and entry point types for contract class
+        // ---------------------------------------------------------
+
+        let path = PathBuf::from("starknet_programs/fibonacci.json");
+        let contract_class = ContractClass::try_from(path).unwrap();
+        let mut entry_points_by_type = HashMap::new();
+
+        let fib_entrypoint_selector = Felt::from_str_radix(
+            "485685360977693822178494178685050472186234432883326654755380582597179924681",
+            10,
+        )
+        .unwrap();
+
+        // insert function fib entrypoint
+        entry_points_by_type.insert(
+            EntryPointType::External,
+            [ContractEntryPoint {
+                selector: fib_entrypoint_selector.clone(),
+                offset: Felt::from_str_radix("33", 10).unwrap(),
+            }]
+            .to_vec(),
+        );
+        let contract_class =
+            ContractClass::new(contract_class.program, entry_points_by_type, None).unwrap();
```

Comment:
@mmsc2 why we are doing this and not directly reading it from the json?

Reply:
yes it should. already fixed it!